src/rule.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file Rule class for use with createRule and withMessage
+ * @author Nick Krause
+ */
+import { Fail } from './fail';
+import { Success } from './success';
+import {
+  CustomValidatorOptions,
+  ValidationCheck,
+  ValidationRule,
+} from './types';
+
+class Rule implements ValidationRule {
+  message: string | ((v: any) => string);
+  opts: CustomValidatorOptions;
+
+  constructor(opts: CustomValidatorOptions) {
+    this.opts = opts;
+  }
+
+  check: ValidationCheck = (value) => {
+    if (this.opts.condition(value)) {
+      // Rule and createRule accept an optional function (delayed) parameter
+      // called transform that allows us to change the value
+      // before it is passed onto the next check.
+      if (this.opts.transform) {
+        return Success.of(this.opts.transform(value));
+      }
+
+      return Success.of(value);
+    } else {
+      if (typeof this.opts.message === 'function') {
```

Comment:
maybe:

`const message = typeof this.opts.message === 'function' ? this.opts.message(value) : this.opts.message;
return Fail.of(message);`